lib/secure-hash.js

```diff
@@ -66,17 +66,43 @@ module.exports = function (base64, crypto, promise) {
 
         config = config || {};
         algorithm = config.algo || "sha512";
-        hashLength = config.hashLength || 48;
+        bytes = config.bytes || 48;
         iterations = config.iterations || 10000;
         salt = new Buffer(config.salt || "", "binary");
 
-        return crypto.pbkdf2Async(data, salt, iterations, hashLength, algorithm).then((result) => {
-            return base64.encodeForUri(result);
+        return crypto.pbkdf2Async(data, salt, iterations, bytes, algorithm);
+    }
+
+
+    function secureHashEncodedAsync(data, config) {
+        return createSecureHashAsync(data, config).then((hash) => {
+            return base64.encode(hash).toString("binary");
+        });
+    }
+
+
+    function secureHashEncodedUriAsync(data, config) {
+        return createSecureHashAsync(data, config).then((hash) => {
+            return base64.encodeForUri(hash);
         });
     }
 
+
+    /**
+     * Hashes data using an algorithm passed in.
+     *
+     * @param {string} data
+     * @param {string} algorithm
+     * @return {Promise.<string>}
+     */
+    function createHash(data, algorithm) {
+        return crypto.createHash(algorithm).update(data).digest("base64");
```

Comment:
I don't think this returns a Promise.  Why does this exist?  It's also not named with *Async at the end of the function's name.